### PR TITLE
ProtectionSquad fixes (Prep Edition)

### PR DESCRIPTION
**Pull request by reaperrr**
Split from #17312 and filed directly against prep.

I don't think #17312 as a whole would be suitable for a hotfix release even if it were finished and merged in time, but these straight-forward fixes are.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs [hunk 1]
```diff
@@ -318,7 +330,7 @@ void ProtectOwn(IBot bot, Actor attacker)
 			{
 				var ownUnits = World.FindActorsInCircle(World.Map.CenterOfCell(GetRandomBaseCenter()), WDist.FromCells(Info.ProtectUnitScanRadius))
 					.Where(unit => unit.Owner == Player && !unit.Info.HasTraitInfo<BuildingInfo>() && !unit.Info.HasTraitInfo<HarvesterInfo>()
-						&& unit.Info.HasTraitInfo<AttackBaseInfo>());
+						&& !unit.Info.HasTraitInfo<AircraftInfo>() && !Info.NavalUnitsTypes.Contains(unit.Info.Name) && unit.Info.HasTraitInfo<AttackBaseInfo>());
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: We have several places (`FindNewUnits`, `CreateAttackForce`, `TryToRushAttack`, `ProtectOwn`) that all use slightly different hardcoded checks to determine unit validity and type.

Mixing hardcoded aircraft checks with yaml-defined naval checks is really ugly, can we please define `AirUnitsTypes` on `SquadManagerBotModuleInfo` to replace the `AircraftInfo` checks?

`ExcludeFromSquadsTypes` is only used in `FindNewUnits`, with the other places using hardcoded checks to exclude harvesters and non-combat units. Can we please define and use `ExcludeFromSquadsTypes` properly?

Sorry for the scope creep, but if we're going to risk touching this now I really think we should do it properly. If you'd rather not then I guess I can try and summon the motivation to do this.

Comment (reaperrr) [reply OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: OK... if you feel like trying go ahead, but if there are any signs of this turning into a can of worms, I'd suggest to not bother (at least for the hotfix release).

Comment (pchote) [reply OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: 5f040597dbb4183d155adc8bcbc583545d159044 shows what I have in mind on the trait side. We'll need to define the actor lists for each mod now.

Comment (pchote) [reply OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: Implementing this properly is too much for the hotfix, so lets drop this for now.

**Comment by pchote** [comment]
Closing, as we (probably) don't want this for prep.

**Pull request closed without merging**